backend/company-agent/src/main/java/org/hyperledger/oa/controller/AdminController.java

```diff
@@ -130,4 +144,60 @@
         return HttpResponse.ok(config);
     }
 
+    /**
+     * Trigger the backend to write configured endpoints to the ledger. TAA digest
+     * has to be passed to explicitly confirm TTA acceptance of the user for this
+     * ledger interaction.
+     * 
+     * @param tAADigest TAA digest retrieved from "/taa/get"
+     * @return {@link HttpResponse}
+     */
+    @Post("/taa/writeEndpoints")
+    public HttpResponse<SchemaAPI> writeEndpoints(@Body String tAADigest) {
+        if (endpointService.isPresent()) {
+            try {
+                Optional<TAAInfo> taa = ac.ledgerTaa();
```

Comment:
I would move all this to the EndpointService, to keep logic away from the controllers. They should only do conversion. Then you also do not need the aries client here. In general there is nothing wrong with your approach, but as the rest of the code follows the principle of keeping controllers small I would like to keep it that way.

Reply:
sounds good, already thought of this but wanted to get your early feedback . check